gc/gc-base/src/test/java/org/projectnessie/gc/base/AbstractRestGC.java

```diff
@@ -99,93 +113,56 @@ void performGc(
+        .set(
+            "spark.sql.extensions",
+            "org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions,org.projectnessie.spark.extensions.NessieSparkSessionExtensions");
 
-  void compareContents(ContentValues actual, ContentValues expected) {
-    if (actual == null && expected == null) {
-      return;
-    }
-    assertThat(actual).isNotNull();
-    assertThat(expected).isNotNull();
-    assertThat(actual.getExpiredContents().size()).isEqualTo(expected.getExpiredContents().size());
-    List<Content> expectedContents = new ArrayList<>(expected.getExpiredContents());
-    expectedContents.sort(new ContentComparator());
-    List<Content> actualContents = new ArrayList<>(actual.getExpiredContents());
-    actualContents.sort(new ContentComparator());
-    for (int i = 0; i < expectedContents.size(); i++) {
-      compareContent(expectedContents.get(i), actualContents.get(i));
-    }
-  }
-
-  void compareContent(Content actual, Content expected) {
-    switch (expected.getType()) {
-      case ICEBERG_TABLE:
-        // exclude global state in comparison as it will change as per the commit and won't be same
-        // as original.
-        // compare only snapshot id for this content.
-        assertThat((IcebergTable) actual)
-            .extracting(IcebergTable::getSnapshotId)
-            .isEqualTo(((IcebergTable) expected).getSnapshotId());
-        break;
-      case ICEBERG_VIEW:
-        // exclude global state in comparison as it will change as per the commit and won't be same
-        // as original.
-        // compare only version id for this content.
-        assertThat((IcebergView) actual)
-            .extracting(IcebergView::getVersionId)
-            .isEqualTo(((IcebergView) expected).getVersionId());
-        break;
-      default:
-        // contents that doesn't have global state can be compared fully.
-        assertThat(actual).isEqualTo(expected);
-    }
+    SparkSession spark =
+        SparkSession.builder()
+            .appName("test-nessie-gc")
+            .master("local[2]")
+            .config(conf)
+            .getOrCreate();
+    spark.sparkContext().setLogLevel("WARN");
+    return spark;
   }
 
-  static class ContentComparator implements Comparator<Content> {
-    @Override
-    public int compare(Content c1, Content c2) {
-      switch (c1.getType()) {
-        case ICEBERG_TABLE:
-          return Long.compare(
-              ((IcebergTable) c1).getSnapshotId(), ((IcebergTable) c2).getSnapshotId());
-        case ICEBERG_VIEW:
-          return Integer.compare(
-              ((IcebergView) c1).getVersionId(), ((IcebergView) c2).getVersionId());
-        default:
-          return c1.getId().compareTo(c2.getId());
-      }
-    }
+  private void verify(
+      Dataset<Row> actual, List<Row> expectedRows, SparkSession session, StructType schema) {
+    Dataset<Row> expected = session.createDataFrame(expectedRows, schema);
```

Comment:
Global note regarding scaling/paralellization. 
I suspect that most often a table has only one ref: `main`. 
In our model, the ref == partition. In that scenario, when we run the job for one table we won't be able to achieve any parallelization. Only one executor will be processing the only ref (`main`) for this table.
@ajantha-bhat Did we think about it?

Reply:
I thought about it before, but I cannot distributively traverse commits. It is currently paginated sequential traversal. 
So, I cannot have a commit level distribution instead of reference level.

May be we need to modify the backend to fetch commit log distributively in future.

Reply:
so currently, in most cases (only `main` branch) the processing won't be distributed at all, isn't it?

Reply:
> so currently, in most cases (only main branch) the processing won't be distributed at all, isn't it?

yes.

But not sure if only main branch is used in most cases. One of the main use case of nessie is branching and tagging.
Main branch can have more commits compared to other branches. But backend doesn't support distributed query of commit logs as of now.